### PR TITLE
chore: user guide for aws sdk v2 signer

### PR DESCRIPTION
**Pull request by matelang**
Signed-off-by: Máté Lang <langmatelaszlo@gmail.com>

### Description
Add documentation about the recently added AWS SDK v2 request signer.

### Issues Resolved
Fixes https://github.com/opensearch-project/opensearch-go/issues/129.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: USER_GUIDE.md [hunk 1]
```diff
@@ -201,3 +216,68 @@ func main() {
 	log.Println("PING OK")
 }
 ```
+
+#### AWS SDK V2
+
+```go
+package main
```
Comment (VachaShah) [inline USER_GUIDE.md:223]: Can we keep the code formatting (like tabs/spaces) for all the examples in this file similar?

Comment (matelang) [reply USER_GUIDE.md:225]: I've reformatted the whole document with standard GoLand formatting options. Hope it's okay.

**Review by VachaShah — DISMISSED** [review]

**Comment by VachaShah** [comment]
Thank you @matelang for adding this! Can you sign off the DCO?

**Comment by matelang** [comment]
> Thank you @matelang for adding this! Can you sign off the DCO?

DCO added.

**Review by VachaShah — APPROVED** [review]

**Pull request merged**